supplier_api_tester/supplier_api_tester/v2/tests/reservation.py

```diff
@@ -234,3 +240,47 @@ def test_reservation(api_url, api_key, product_id, version=2):
             response=raw_response,
         )
     return TestResult()
+
+
+@test_wrapper
+def test_reservation_with_unit_prices(api_url, api_key, product_id, version=2):
+    """Testing reservation for product with provide_pricing=True"""
+    catalog_response: Response
+    products: List[Product]
+    catalog_response, products = get_catalog(api_url, api_key, version)
+
+    if not product_provides_pricing(product_id, products):
+        return TestResult(
+            status=1,
+            message=(
+                "Skipping the test because the product does not provide pricing."
+            )
+        )
+
+    url = f'{api_url}/v{version}/products/{product_id}/reservation'
+    slot = get_reservation_slot(api_url, api_key, product_id)
+    json_payload: Dict = get_payload_for_reservation(api_url, api_key, product_id, slot)
+    raw_response, response = client(url, api_key, method=requests.post, json_payload=json_payload)
+    reservation: Reservation = get_reservation(raw_response, response)
+
+    # we expect the reservation response to have unit_price
+    if not reservation.unit_price:
+        raise FailedTest(
+            message=f'Product {product_id} provides pricing but the response does not include unit_price.',
+            response=raw_response,
+        )
+
+    for variant_id in set(ticket.get('variant_id') for ticket in json_payload.get('tickets')):
+        if variant_id not in reservation.unit_price:
+            raise FailedTest(
+                message=f'Product {product_id} provides pricing but the response is missing unit price for a variant',
+                response=raw_response,
+            )
+
+        if not reservation.unit_price.get(variant_id).currency or not reservation.unit_price.get(variant_id).amount:
```

Comment:
I think it would be enough `if not reservation.unit_price.get(variant_id)` as if there is the variant_id the `VariantPrice` is not optional

Reply:
I think it's checking if at least one property is missing. It's possible that we get an amount but not the currency.

Reply:
you mean if they are not empty. right, then it makes sense :+1: